src/Form/matching-config/matching-config-source.js

```diff
@@ -278,7 +278,7 @@ const matchingConfiguration = {
                 },
                 username: {
                     match:
-                        '(user|account|online.?id|membership.?id|log(i|o)n|net)((.)?(name|i.?d.?|log(i|o)n).?)?(.?((or|/).+|\\*|:)( required)?)?$' +
+                        '(user|account|online.?id|membership.?id|log(i|o)n|net|customer.?number)((.)?(name|i.?d.?|log(i|o)n).?)?(.?((or|/).+|\\*|:)( required)?)?$' +
```

Comment:
```suggestion
                        '(user|account|online.?id|membership.?id|log(i|o)n|net|customer.?(number|id))((.)?(name|i.?d.?|log(i|o)n).?)?(.?((or|/).+|\\*|:)( required)?)?$' +
```

Just a suggestion to add `customer id` as well. TBH, this regex is totally messed up at this point. We are repeating the `id` 4 times, and that `required` at the end is total madness. I think now we have better tools for this sort of thing and we should probably revisit this regex.